src/views/UsersPage.vue

```diff
@@ -1,42 +1,132 @@
+
+type Canvases = typeof authStore.canvases
+
+const name = ref('')
+const icon = ref('')
+const profile = ref('')
+const canvases: Canvases = ref({})
+const isValidUser = ref()
+
+onMounted(() => {
+  setProfile()
+})
+
+watch(route, () => {
+  initializeProfile()
+  setProfile()
+})
+
+const setProfile = () => {
+  const otherUserUID = String(route.params.user_id)
+  if (authStore.uid !== otherUserUID) {
+    getDoc(doc(db, 'users', otherUserUID))
+      .then((userDocSnap) => {
+        if (userDocSnap.exists()) {
+          isValidUser.value = true
+          name.value = userDocSnap.data().name
+          icon.value =
+            userDocSnap.data().icon === '' ? Icon : userDocSnap.data().icon
+          profile.value = userDocSnap.data().profile
+          // canvas取得
+          setCanvases(otherUserUID)
+        } else {
+          isValidUser.value = false
+          canvases.value = {}
+          console.log('プロフィール情報が見つかりません')
+        }
+      })
+      .catch((error) => {
+        console.log('プロフィール情報の取得に失敗しました：', error)
+      })
+  } else {
+    isValidUser.value = true
+    icon.value = authStore.icon
+    profile.value = authStore.profile
+    canvases.value = authStore.canvases
+  }
+}
+
+const setCanvases = (otherUserUID: string) => {
+  const canvasQuery = query(
+    collection(db, 'canvas'),
+    where('uid', '==', otherUserUID),
+    where('isShare', '==', true),
+  )
+  getDocs(canvasQuery)
+    .then((querySnapshot) => {
+      querySnapshot.forEach((document) => {
+        const canvasID = document.id
+        canvases[canvasID] = document.data()
+      })
+    })
+    .catch((error) => {
+      console.log(error)
+    })
+}
+
+const initializeProfile = () => {
+  icon.value = ''
+  profile.value = ''
+  canvases.value = ''
+}
+
+const userCol = () => {
+  if (authStore.uid === route.params.user_id) {
+    return 'col-span-12 sm:col-span-9 sm:ml-8'
+  }
+  return 'sm:mx-8'
+}
 </script>
 
 <template lang="pug">
 div(class="grid grid-cols-12 mt-4 sm:mt-8")
   //- left
-  div(class="col-span-3 lg:col-span-2 sm:block hidden")
+  div(v-if="authStore.uid === route.params.user_id" class="col-span-3 lg:col-span-2 sm:block hidden")
     div(class="border rounded-lg p-4")
       ul(class="text-midnightBlue")
         li(class="border-b pb-2")
           router-link(to="users" class="text-dustyOrange block") ユーザー情報
         //- li(class="border-b py-2")
-        //-   a(href="#" class="block") ブックマーク
+        //-   a(href="#" class="block") お気に入り
         //- li(class="border-b py-2")
         //-   a(href="#" class="block") 設定
         li(class="pt-2")
           button(class="block" @click="authStore.logout()") ログアウト
   //- right
-  div(class="col-span-12 sm:col-span-9 sm:ml-8")
+  div(:class="userCol()" class="col-span-12")
     //- top
-    div(class="relative sm:static sm:grid sm:grid-cols-12 pb-4 border-b")
+    div(v-if="isValidUser===true" class="relative sm:static sm:grid sm:grid-cols-12 pb-4 border-b")
```

Comment:
```suggestion
    div(v-if="isValidUser" class="relative sm:static sm:grid sm:grid-cols-12 pb-4 border-b")
```
個人的にはbooleanなら isValidUser === tureにしなくてもいいのかなと思いますが、見やすさの問題ですかね。

Reply:
`===` にしなくて良いですね

相談です。今回表示パターンが3つあります。
- ユーザー有無を判断中 → 何も表示しない
- ユーザーが存在する → profile表示
- ユーザーが存在しない → 「このアカウントは存在しません」と表示

パターンをtrue/falseで行っているのですが、文字列の方が良いですかね？
指摘の箇所は `if="isValidUser"` で意図した通りに表示されるのですが、
`else-if="!isValidUser"` にすると、「このアカウントは存在しません」と一瞬表示されます。

Reply:
別スレッドで案を書いてみました。